src/main/java/com/cburch/logisim/analyze/model/TruthTable.java

```diff
@@ -664,11 +664,10 @@ private void removeInput(int index, int oldCount) {
       int b = (1 << (oldCount - 1 - index)); // _0001000
       boolean[] changed = new boolean[columns.size()];
       // loop rows by index to avoid java.util.ConcurrentModificationException
-      for (int i = 0; i < rows.size(); ++i) {
-        Row r = rows.get(i);
-        if (r.inputs[index] == Entry.DONT_CARE)
-          continue;
-        setDontCare(r, b, true, changed); // mutates row
+      for (Row r : rows) {
```

Comment:
maybe have a look at the comment at line 666

Reply:
Ough, impressive I missed that mutation. Fixed.